pdks/SKY130_PDK/primitive.py

```diff
@@ -0,0 +1,457 @@
+from align.primitive import default
+
+#files copied from default primitives
+
+#############################################################
+#primitives for SKY130_PDK (contents copied from OSU180)
+#############################################################
+
+# So basically, there are 2 PR's (pull requests)
+#1. Installation
+#2. Porting ALIGN to work for OSU180
+
+#Installation seems to working fine.
+#Porting it to 180 seems to be a challenge as of now. We are working in that direction.
+
+
+
+# I do not think OSU180 has various varieties of VT cells.
+#Hence I commented out the LVT & HVT part
+
+# I am not quite sure how the self.m2PerUnitCell function works. It is calculating with respect to the "fin" no. and size.
+#I think we need to change the method signature to fit our OSU180 PDK.
+
+
+# I have not run this using ALIGN. I think we need to have a meeting to check the correctness of this file.
+
+#We are only copying code from align/primitive/default/mos.py
+#what about via.py/canvas.py?
+
+
+#In the layers.json file "Fin" is mapped to "active". Is this how it is supposed to be?
+#I am having some doubts here.
+
+
+
+
+
+
+class MOSGenerator(default.MOSGenerator):
```

Comment:
Don't inherit from default.MOSGenerator since you are effectively reimplementing everything in it.

Can you summarize what changes you need to make from a layer perspective? Its also possible that simply changing layers.json may be enough.

Reply:
@arvuce22 can help with this exercise too.